### PR TITLE
[BUGFIX beta] Router#url should not error when `location` is a string

### PR DESCRIPTION
**Pull request by ryanolsonx**
See: #18260

### CODE REVIEW

File: packages/@ember/-internals/routing/lib/system/router.ts [hunk 1]
```diff
@@ -1791,7 +1791,13 @@ EmberRouter.reopen(Evented, {
    @private
  */
   url: computed(function(this: Router<Route>) {
-    return get(this, 'location').getURL();
+    let location = get(this, 'location');
+
+    if (typeof location !== 'string') {
```
Comment (btecu) [inline packages/@ember/-internals/routing/lib/system/router.ts]: Could avoid the negation by swapping the returns

Comment (ryanolsonx) [reply packages/@ember/-internals/routing/lib/system/router.ts]: That's a good idea.  I'll do that.

**Review by rwjblue — APPROVED** [review]
Awesome, thank you!

**Comment by rwjblue** [comment]
Hmm, I've restarted this CI job a couple of times and keep seeing the same failure.

**Comment by rwjblue** [comment]
```
not ok 25 IE 11.0 - [35 ms] - {{on}} Modifier:  asserts if the provided callback accesses `this` without being bound prior to passing to on
    ---
        actual: >
            [object HTMLButtonElement]
        expected: >
            null
        stack: >
               at myFunc (http://127.0.0.1:7774/200750294740/dist/tests/ember-tests.js:41220:13)
        message: >
            this is null on browsers without native proxy support
        negative: >
            false
        browser log: |
    ...
```

**Comment by rwjblue** [comment]
I can't figure out how that failure could be related to the changes here. 🤔

I think we need to get a test run in an actual IE11 box to see whats really going on (why does it pass on master, why does this seemingly unrelated change make it fail, etc).

**Comment by ryanolsonx** [comment]
Yeah - that's pretty weird.  I rebased on master and hopefully things just pass this next time around.

**Comment by rwjblue** [comment]
Yep, that fixed things up for us. Thank you again @ryanolsonx!

**Pull request merged**